lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -511,7 +489,15 @@ export class PublicClientApplication {
      *
      */
     public getRedirectUri(): string {
-        return this.authModule.getRedirectUri();
+        if (this.config.auth.redirectUri) {
+            if (typeof this.config.auth.redirectUri === "function") {
+                return this.config.auth.redirectUri();
+            } else if (!StringUtils.isEmpty(this.config.auth.redirectUri)) {
```

Comment:
I would like to better understand the `redirectUri` as a function use case, as it would be nice to remove if its not truly needed. Needing code like this is annoying, frankly.

Reply:
Also, this statement will never be false, given the truthy check above.

Reply:
I believe we can drop the redirectUri as a function, and just adjust the way to select the current location as the default.

```js
public getRedirectUri(): string {
  return this.config.auth.redirectUri || window.location.href;
}
```

And then make sure we don't set `window.location.href` as the redirectUri when `PublicClientApplication` is instantiated.

Reply:
will follow up in the next PR